### PR TITLE
kl-ps-welcome-prompt

### LINKED ISSUE #7: 7. As a user, I want to see a welcoming prompt to add my first item if my list is empty to help me get oriented to how the app works.
## Summary

When a user starts a new list, they need a little guidance on how to get started. Think through what you'd want to see as a user and make some intentional user experience decisions that will guide our users in confidently adding their very first item. 

## Acceptance criteria
- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Notes
- [The wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-empty.png) is a little short on the "welcoming" part of the story, but otherwise is functionally complete for the purposes of the AC.

### PR DESCRIPTION
**Pull request by snowsneakers**
## Description

This code adds a welcome message to the list page if the list is empty. We used Link from react-router-dom to navigate the user to the /add-item page. We learned that an `a` tag will trigger a page refresh and `Link` won't. This allows state to persist through page navigation. We also fixed an issue where the string 'null' was being added as the token in local storage instead of `null`.

## Related Issue

Closes #7 

## Acceptance Criteria

- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓    | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

issue-7-before

### After

issue-7-after

## Testing Steps / QA Criteria

1. Clear local storage to clear list
2. Welcome message should appear on list page
3. Click button should navigate to add item page

### CODE REVIEW

**Comment by NuriyaAkh** [comment]
Nice work!
 if the user tries to add the first item without creating the list, an error occurs
 
it is not clear to users if they need to create a new list before adding the first item

**Review by NuriyaAkh — APPROVED** [review]

**Review by meera-ramesh19 — APPROVED** [review]
Looks good

**Review by georgeoffley — APPROVED** [review]
Looks awesome. Nice work!!

File: src/views/List.jsx [hunk 1]
```diff
@@ -11,29 +12,35 @@ export function List({ data }) {
...
-				{data
-					.filter((item) =>
-						item.name.toLowerCase().includes(searchQuery.toLowerCase()),
-					)
-					.map((item) => {
-						return <ListItem key={item.id} name={item.name} />;
-					})}
-			</ul>
+			{data.length > 0 ? (
+				<ul>
```
Comment (cafloyd) [inline src/views/List.jsx:16]: Curious about the reasoning for moving this `<ul>` from coming after the `<form>` to now wrapping the `<form>`. I might be missing something, but it seems like it makes sense to keep it after the `<form>`.

**Review by cafloyd — APPROVED** [review]
Nice work, y'all! Just left one small comment for y'all to address - if you have a reason for the change I noted, don't be afraid to argue your case! But otherwise, let's rearrange those HTML elements! Approving this PR with the assumption that you'll either explain your reasoning or make that change before merging!

File: src/views/List.jsx (hunk 1 above)
Comment (snowsneakers) [reply src/views/List.jsx]: We overlooked that. I updated it

**Pull request merged**